"panic: runtime error: close of closed channel" occurs when inputting long query

CentOS 上でこのバイナリを使っていて起こりました。
https://github.com/peco/peco/releases/download/v0.1.9/peco_linux_amd64.tar.gz

再現条件がよくわかってないですが、 `panic: runtime error: close of closed channel` となって異常終了することがあるようです。

長いクエリを入力していると発生するように見えます。
`a` のあとに半角スペースを入力し続けたりしていると発生しますが、すぐ出るときとなかなか出ないときとあります。

以下はそのときのエラー出力です。
ただ、これが発生するとターミナルの出力が壊れてしまうので、適当に整形して貼り付けています。
http://gyazo.com/a92f93cb90b1e1ce8c44858a687c2788

```
panic: runtime error: close of closed channel

goroutine 503 [running]:
runtime.panic(0x59d660, 0x6c72b5)
/usr/local/go/src/pkg/runtime/panic.c:279 +0xf5
github.com/peco/peco.(*RegexpMatcher).Match(0xc2080226c0, 0xc2080785a0, 0xc208192f60, 0xe, 0xc20817a000, 0x3e1, 0x3e1, 0x0, 0x0, 0x0)
/home/ubuntu/go/src/github.com/peco/peco/matchers.go:259 +0x26e
github.com/peco/peco.(*Filter).Work(0xc208000730, 0xc2080785a0, 0xc208192f60, 0xe)
/home/ubuntu/go/src/github.com/peco/peco/filter.go:13 +0x270
created by github.com/peco/peco.(*Filter).Loop
/home/ubuntu/go/src/github.com/peco/peco/filter.go:39 +0x215

goroutine 16 [semacquire]:
sync.runtime_Semacquire(0xc20800076c)
/tmp/makerelease197226928/go/src/pkg/runtime/sema.goc:199 +0x30
sync.(*WaitGroup).Wait(0xc20803e240)
/usr/local/go/src/pkg/sync/waitgroup.go:129 +0x14b
github.com/peco/peco.(*Ctx).WaitDone(0xc208050300)
/home/ubuntu/go/src/github.com/peco/peco/ctx.go:134 +0x2e
main.main()
/home/ubuntu/go/src/github.com/peco/peco/cmd/peco/peco.go:170 +0xea1

goroutine 19 [finalizer wait]:
runtime.park(0x414820, 0x6cb400, 0x6c9f49)
/usr/local/go/src/pkg/runtime/proc.c:1369 +0x89
runtime.parkunlock(0x6cb400, 0x6c9f49)
/usr/local/go/src/pkg/runtime/proc.c:1385 +0x3b
runfinq()
/usr/local/go/src/pkg/runtime/mgc0.c:2644 +0xcf
runtime.goexit()
/usr/local/go/src/pkg/runtime/proc.c:1445

goroutine 20 [syscall]:
os/signal.loop()
/usr/local/go/src/pkg/os/signal/signal_unix.go:21 +0x1e
created by os/signal.init·1
/usr/local/go/src/pkg/os/signal/signal_unix.go:27 +0x32

goroutine 17 [syscall]:
runtime.goexit()
/usr/local/go/src/pkg/runtime/proc.c:1445

goroutine 21 [select]:
github.com/nsf/termbox-go.func·001()
/home/ubuntu/go/src/github.com/nsf/termbox-go/api.go:86 +0x41a
created by github.com/nsf/termbox-go.Init
/home/ubuntu/go/src/github.com/nsf/termbox-go/api.go:105 +0x82e

goroutine 22 [runnable]:
github.com/peco/peco.(*View).Loop(0xc208039f48)
/home/ubuntu/go/src/github.com/peco/peco/view.go:34 +0x1aa
created by main.main
/home/ubuntu/go/src/github.com/peco/peco/cmd/peco/peco.go:158 +0xd34

goroutine 23 [select]:
github.com/peco/peco.(*Filter).Loop(0xc208000730)
/home/ubuntu/go/src/github.com/peco/peco/filter.go:27 +0x239
created by main.main
/home/ubuntu/go/src/github.com/peco/peco/cmd/peco/peco.go:159 +0xd4f

goroutine 24 [select]:
github.com/peco/peco.(*Input).Loop(0xc208000740)
/home/ubuntu/go/src/github.com/peco/peco/input.go:37 +0x431
created by main.main
/home/ubuntu/go/src/github.com/peco/peco/cmd/peco/peco.go:160 +0xd6a

goroutine 25 [select]:
github.com/peco/peco.(*Ctx).SignalHandlerLoop(0xc208050300, 0xc208029110)
/home/ubuntu/go/src/github.com/peco/peco/ctx.go:238 +0xe3
created by main.main
/home/ubuntu/go/src/github.com/peco/peco/cmd/peco/peco.go:161 +0xd92

goroutine 26 [select]:
github.com/nsf/termbox-go.PollEvent(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
/home/ubuntu/go/src/github.com/nsf/termbox-go/api.go:244 +0x396
github.com/peco/peco.func·001()
/home/ubuntu/go/src/github.com/peco/peco/input.go:30 +0x27
created by github.com/peco/peco.(*Input).Loop
/home/ubuntu/go/src/github.com/peco/peco/input.go:32 +0xce

goroutine 504 [runnable]:
github.com/peco/peco.(*RegexpMatcher).MatchAllRegexps(0xc2080226c0, 0xc2080ecd80, 0xc, 0x10, 0xc2080236b0, 0x30, 0x0, 0x0, 0x0)
/home/ubuntu/go/src/github.com/peco/peco/matchers.go:296 +0x455
github.com/peco/peco.func·004()
/home/ubuntu/go/src/github.com/peco/peco/matchers.go:243 +0x152
created by github.com/peco/peco.(*RegexpMatcher).Match
/home/ubuntu/go/src/github.com/peco/peco/matchers.go:250 +0x203

goroutine 498 [runnable]:
regexp.(*Regexp).FindAllStringSubmatchIndex(0xc2082cd7c0, 0xc2080435c0, 0x38, 0xffffffffffffffff, 0xc208024800, 0x39, 0x55)
/usr/local/go/src/pkg/regexp/regexp.go:1059
github.com/peco/peco.(*RegexpMatcher).MatchAllRegexps(0xc2080226c0, 0xc2080eca00, 0xc, 0x10, 0xc2080435c0, 0x38, 0x0, 0x0, 0x0)
/home/ubuntu/go/src/github.com/peco/peco/matchers.go:279 +0x120
github.com/peco/peco.func·004()
/home/ubuntu/go/src/github.com/peco/peco/matchers.go:243 +0x152
created by github.com/peco/peco.
/home/ubuntu/go/src/github.com/peco/peco/matchers.go:250 +0x203

goroutine 493 [runnable]:
regexp.(*Regexp).FindAllStringSubmatchIndex(0xc2082cdd60, 0xc208096180, 0x2e, 0xffffffffffffffff, 0xc208025000, 0x2f, 0x55)
/usr/local/go/src/pkg/regexp/regexp.go:1059
github.com/peco/peco.(*RegexpMatcher).MatchAllRegexps(0xc2080226c0, 0xc2080ec180, 0xc, 0x10, 0xc208096180, 0x2e, 0x0, 0x0, 0x0)
/home/ubuntu/go/src/github.com/peco/peco/matchers.go:279 +0x120
github.com/peco/peco.func·004()
/home/ubuntu/go/src/github.com/peco/peco/matchers.go:243 +0x152
created by github.com/peco/peco.(*RegexpMatcher).Match
/home/ubuntu/go/src/github.com/peco/peco/matchers.go:250 +0x203

goroutine 494 [runnable]:
regexp.(*Regexp).FindAllStringSubmatchIndex(0xc20803b7c0, 0xc2080432c0, 0x3b, 0xffffffffffffffff, 0xc208100800, 0x3c, 0x55)
/usr/local/go/src/pkg/regexp/regexp.go:1059
github.com/peco/peco.(*RegexpMatcher).MatchAllRegexps(0xc2080226c0, 0xc2080ecb00, 0xc, 0x10, 0xc2080432c0, 0x3b, 0x0, 0x0, 0x0)
/home/ubuntu/go/src/github.com/peco/peco/matchers.go:279 +0x120
github.com/peco/peco.func·004()
/home/ubuntu/go/src/github.com/peco/peco/matchers.go:243 +0x152
created by github.com/peco/peco.(*RegexpMatcher).Match
/home/ubuntu/go/src/github.com/peco/peco/matchers.go:250 +0x203

goroutine 495 [runnable]:
github.com/peco/peco.(*Filter).Work(0xc208000730, 0xc208078300, 0xc2082644a0, 0xd)
/home/ubuntu/go/src/github.com/peco/peco/filter.go:8
created by github.com/peco/peco.(*Filter).Loop
/home/ubuntu/go/src/github.com/peco/peco/filter.go:39 +0x215
```

--- a/matchers.go
+++ b/matchers.go
@@ -265,7 +265,7 @@ MATCH:
 			// so we punt it by letting the close() happen at a separate
 			// goroutine, protected by a defer recover()
 			go func() {
-				defer recover()
+				defer func() { recover() }()
 				close(iter)
 			}()
 			break MATCH

